api/rbac/v1alpha1/annotations.go

```diff
@@ -30,3 +45,28 @@ func OIDCClaimNameFromAnnotationKey(key string) (string, bool) {
 	}
 	return strings.TrimPrefix(key, AnnotationKeyOIDCClaimNamePrefix), true
 }
+
+// OIDCClaimsFromAnnotationValue parses the value of an
+// rbac.kargo.akuity.io/claims annotation and returns the corresponding list of
+// formatted and sorted claims.
+func OIDCClaimsFromAnnotationValue(value string) ([]string, error) {
+	claims := make(map[string][]string)
+	if err := json.Unmarshal([]byte(value), &claims); err != nil {
+		return nil, fmt.Errorf("unmarshaling OIDC claims from annotation value: %w", err)
+	}
+	var result []string
+	for name, values := range claims {
+		for _, v := range values {
+			result = append(result,
+				FormatClaim(name, strings.TrimSpace(v)),
```

Comment:
This seems as if it's not right. This is going to return results in a way specially formatted for use with the index we've built of SAs by claim. Other than building or using that index, I don't think there is any other context where we want claims to look like this, and I believe I see this reflected in some of the tests where you are expecting values like `email/kilgore@kilgore.trout` where we ought not to be.